cmd/spire-server/cli/federation/common.go

```diff
@@ -1,9 +1,160 @@
+	BundleEndpointProfile string          `json:"bundle_endpoint_profile,omitempty"`
+	EndpointSPIFFEID      string          `json:"endpoint_spiffe_id,omitempty"`
+	Bundle                json.RawMessage `json:"bundle,omitempty"`
+	BundleFormat          string          `json:"bundle_format,omitempty"`
+	BundlePath            string
+}
+
+// federationRelationshipsFromFile parse a json file into types FederationRelationships
+func federationRelationshipsFromFile(path string) ([]*types.FederationRelationship, error) {
+	r := os.Stdin
+	if path != "-" {
+		f, err := os.Open(path)
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
+		r = f
+	}
+
+	dat, err := io.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+
+	relationships := &federationRelationships{}
+	if err := json.Unmarshal(dat, &relationships); err != nil {
+		return nil, fmt.Errorf("failed to parse JSON: %w", err)
+	}
+
+	protoRelationships := []*types.FederationRelationship{}
+	for i, relationship := range relationships.FederationRelationships {
+		protoRelationship, err := jsonToProto(relationship)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse item %d: %w", i, err)
+		}
+		protoRelationships = append(protoRelationships, protoRelationship)
+	}
+
+	return protoRelationships, nil
+}
+
+func jsonToProto(fr *federationRelationshipJSON) (*types.FederationRelationship, error) {
+	if fr.TrustDomain == "" {
+		return nil, errors.New("trust domain is required")
+	}
+
+	if fr.BundleEndpointURL == "" {
+		return nil, errors.New("bundle endpoint URL is required")
+	}
+
+	proto := &types.FederationRelationship{
+		TrustDomain:       fr.TrustDomain,
+		BundleEndpointUrl: fr.BundleEndpointURL,
+	}
+
+	switch fr.BundleEndpointProfile {
+	case profileHTTPSWeb:
+		if fr.EndpointSPIFFEID != "" {
+			return nil, errors.New("the 'https_web' endpoint profile does not expect an endpoint SPIFFE ID")
+		}
+		if fr.BundlePath != "" {
+			return nil, errors.New("the 'https_web' endpoint profile does not expect a bundle")
+		}
+		proto.BundleEndpointProfile = &types.FederationRelationship_HttpsWeb{
+			HttpsWeb: &types.HTTPSWebProfile{},
+		}
+
+	case profileHTTPSSPIFFE:
+		if fr.EndpointSPIFFEID == "" {
+			return nil, errors.New("endpoint SPIFFE ID is required if 'https_spiffe' endpoint profile is set")
+		}
+		endpointSPIFFEID, err := spiffeid.FromString(fr.EndpointSPIFFEID)
+		if err != nil {
+			return nil, fmt.Errorf("cannot parse bundle endpoint SPIFFE ID: %w", err)
+		}
+
+		var bundle *types.Bundle
+		switch {
+		case fr.BundlePath != "":
+			bundle, err = bundleFromPath(fr.BundlePath, fr.BundleFormat, endpointSPIFFEID.TrustDomain().String())
+			if err != nil {
+				return nil, err
+			}
+		case fr.Bundle != nil:
+			bundle, err = bundleFromRawMesssage(fr.Bundle, fr.BundleFormat, endpointSPIFFEID.TrustDomain().String())
+			if err != nil {
+				return nil, fmt.Errorf("cannot parse bundle raw message: %w", err)
+			}
+		}
```

Comment:
We should allow the `TrustDomainBundle` to be set independent of the profile being used.